tests/client/crud_spec/framework.rs

```diff
@@ -0,0 +1,78 @@
+macro_rules! run_find_test {
+    ( $db:expr, $c:expr, $f:expr, $o:expr, $t:expr ) => {{
+        let mut cursor = $c.find($f, $o).unwrap();
+
+        let array = match $t.result {
+            Bson::Array(ref arr) => arr.clone(),
+            _ => panic!("Invalid `result` of find test")
+        };
+
+        for bson in array {
+            assert!(eq::bson_eq(&bson, &Bson::Document(cursor.next().unwrap())));
+        }
+
+        assert!(!cursor.has_next());
+        check_coll!($db, $c, $t.collection);
+    }};
+}
+
+macro_rules! run_insert_one_test {
+    ( $db:expr, $c: expr, $d:expr, $o:expr) => {{
+        let inserted = $c.insert_one($d, None).unwrap().inserted_id.unwrap();
+        let id = match $o.result {
+            Bson::Document(ref doc) => doc.get("insertedId").unwrap(),
+            _ => panic!("`insert_one` test result should be a document")
+        };
+
+        assert!(eq::bson_eq(&id, &inserted));
+        check_coll!($db, $c, $o.collection);
+    }};
+}
+
+
+#[macro_export]
+macro_rules! run_suite {
+    ( $f:expr, $c:expr ) => {{
```

Comment:
Let's avoid 1-char variable names, even for macros. Could we write `$file` instead?